diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt

```diff
@@ -647,8 +653,209 @@ class NewlinesRuleWarnTest : LintTestBase(::NewlinesRule) {
+    fun `should warn dot qualified with exclexcl`() {
+        lintMethod(
+                """
+                |fun foo() {
+                |   x!!.map()
+                |   .gre()
+                |   .few()
+                |}
+                |
+                |fun foo() {
+                |   x!!
+                |   .map()
+                |   .gre()
+                |   .few()
+                |}
+                |
+                |fun foo() {
+                |   x!!
+                |   .map().gre()
+                |   .few()
+                |}
+                |
+                |fun foo() {
+                |   x!!.map()
+                |   .gre().few()
+                |}
+            """.trimMargin(),
+                LintError(2, 7, ruleId, "$functionalStyleWarn .", true),
+                LintError(16, 10, ruleId, "$functionalStyleWarn .", true),
+                LintError(21, 7, ruleId, "$functionalStyleWarn .", true),
+                LintError(22, 10, ruleId, "$functionalStyleWarn .", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `should warn elvis`() {
+        lintMethod(
+                """
+                |fun foo() {
+                |
+                |   z.goo()
+                |       ?:
+                |        goo()
+                |      
+                |   x.goo()
+                |       ?:goo()
+                |       
+                |   y.ds()?:gh()
+                |}
+            """.trimMargin(),
+                LintError(4,8, ruleId, "$shouldBreakBefore ?:", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `should warn elvis with several dot qualifided`() {
+        lintMethod(
+                """
+                |fun foo() {
+                |   z.goo()
+                |       ?:
+                |        goo().gor().goo()
+                |}
+            """.trimMargin(),
+                LintError(3,8, ruleId, "$shouldBreakBefore ?:", true),
+                LintError(4,20, ruleId, "$functionalStyleWarn .", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `test configuration for calls in one line`() {
+        lintMethod(
+                """
+                |fun foo() {
+                |   z.goo().foo().qwe()
+                |   z!!.htr().foo()
+                |   x.goo().foo().goo()
+                |   x.gf().gfh() ?: true
+                |   x.gf().fge().qwe().fd()
+                |}
+            """.trimMargin(),
+                LintError(6,22,ruleId, "$functionalStyleWarn .", true),
+                rulesConfigList = rulesConfigList
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `more test for prefix`() {
+        lintMethod(
+                """
+                |fun foo() {
+                |   foo
+                |       .bar()
+                |       .goo()
+                |       .qwe()!!
```

Comment:
Please add the test where `!!` is used in the middle of the chain:
```kotlin
foo
    .bar()
    .baz()!!
    .qwe()
```
It shouldn't emit warnings, but currently emits a very strange one (`should break a line before and not after foo...`).